pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/embeddings/OpenAIEmbeddingsService.java

```diff
@@ -0,0 +1,30 @@
+package com.datastax.oss.pulsar.functions.transforms.embeddings;
+
+import com.azure.ai.openai.OpenAIClient;
+import com.azure.ai.openai.models.Embeddings;
+import com.azure.ai.openai.models.EmbeddingsOptions;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class OpenAIEmbeddingsService implements EmbeddingsService {
+
+  private final OpenAIClient openAIClient;
+  private final String model;
+
+  public OpenAIEmbeddingsService(OpenAIClient openAIClient, String model) {
+    this.openAIClient = openAIClient;
+    this.model = model;
+  }
+
+  @Override
+  public List<List<Double>> calculateEmbeddings(List<String> texts) {
+    EmbeddingsOptions embeddingsOptions = new EmbeddingsOptions(texts);
```

Comment:
we should add metrics one day here

Reply:
good point, we can do it in a follow up pull request. Currently there's no metric for any transformations. 
we could add a generic implementation per-step and let the step adds more specific metrics 

Unfortuantely the current Pulsar Functions Framework only supports counters. In this case it would be great to measure latency/thoughrput of the openAI API calls